dali/pipeline/executor/queue_policy.h

```diff
@@ -215,14 +236,18 @@ struct SeparateQueuePolicy {
     return result;
   }
 
-  void ReleaseIdxs(OpType stage, QueueIdxs idxs) {
+  void ReleaseIdxs(OpType stage, QueueIdxs idxs, cudaStream_t stage_stream = 0) {
+    if (idxs[stage] == kInvalidIdx) {
+      return;
+    }
     int current_stage = static_cast<int>(stage);
-    // We have a special case for Support ops - they are set free by a GPU stage,
-    // during QueueOutputIdxs
-    if (stage != OpType::CPU) {
-      if (HasPreviousStage(stage)) {
-        ReleaseStageIdx(PreviousStage(stage), idxs);
-      }
+    // TODO(klecki) when we move to CUDA 10, we should move to cudaLaunchHostFunc
```

Comment:
Maybe ifdef for CUDA 10? We have CUDA 10 already.

Reply:
This API is described as to be deprecated in some future, but is still available in both CUDA 9 and 10. Maybe it is better to not have build dependent code paths until we actually need to.